### ISSUE TITLE
Windows service cannot be stopped

### ISSUE THREAD

**Issue by tavrez** [cloudflare/cloudflared#82]
Hello, I've successfully installed version 2019.4 of cloudflared.exe as a windows service and it's working perfectly, however when I want to stop the service, it get stucked on "stopping" state and there is no way to terminate it, could you please take a look?

My windows is version 16299.1029

**Comment by sssilver** [comment]
Hey, thanks for reporting this!

Are you trying to stop the service through the Windows Services management pane, or sc.exe?

**Comment by tavrez** [comment]
I tried both, they send service into stopping state, but it goes forever.

**Comment by sssilver** [comment]
Would you be able to provide `cloudflared` logs, and relevant pieces of the Windows Event Log  (log name: Application, Source: Cloudflared)?

**Comment by tavrez** [comment]
Windows Event Log only shows "received stop control request" when I click stop and it will stay like that forever.
I don't know where `cloudflared` log is

**Comment by tavrez** [comment]
I updated into the last version, still same problem.
I also upgraded my Windows into 1809, no difference.

**Comment by cold-flirtz** [comment]
![image](https://user-images.githubusercontent.com/55838314/65686087-c0226180-e08d-11e9-9005-44e88f102019.png)
same here, please tell me the cloudflared log location so I can provide that.

**Comment by thymbahutymba** [comment]
Same issue here. `cloudflared version 2019.9.2 (built 2019-09-26-1916 UTC)` 
The only way to stop it is to kill it with PowerShell.

**Comment by jmuto** [comment]
Same issue here.

Windows 10 Pro 1903 (18362.356)
cloudflared version 2019.9.2 (built 2019-09-26-1916 UTC)
`C:\Windows\system32\config\systemprofile\.cloudflared\config.yml` :
```
proxy-dns: true
proxy-dns-upstream:
 - https://1.1.1.1/dns-query
 - https://1.0.0.1/dns-query
origincert: C:\Windows\system32\config\systemprofile\.cloudflared\cert.pem
```

1. Type on admin command prompt.
```
"C:\Program Files\cloudflared\cloudflared.exe" service install
```

2. Edit registry to write logfile, `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Cloudflared\ImagePath` .
```
"C:\Program Files\cloudflared\cloudflared.exe" --loglevel debug --logfile C:\ProgramData\cloudflared.log
```

3. Type on admin command prompt.
```
sc start cloudflared
```

4. Type on admin command prompt.
```
sc stop cloudflared
```

5. cloudflared (Argo Tunnel agent) cannot be stopped.
```

SERVICE_NAME: cloudflared
        TYPE               : 10  WIN32_OWN_PROCESS
        STATE              : 3  STOP_PENDING
                                (NOT_STOPPABLE, NOT_PAUSABLE, IGNORES_SHUTDOWN)
        WIN32_EXIT_CODE    : 0  (0x0)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x0
```

6. The logfile does not contain messages about the problem.
```
{"level":"info","msg":"Version 2019.9.2","time":"2019-09-29T22:44:56+09:00"}
{"level":"info","msg":"GOOS: windows, GOVersion: go1.12.7, GoArch: amd64","time":"2019-09-29T22:44:56+09:00"}
{"level":"info","logfile":"C:\\ProgramData\\cloudflared.log","loglevel":"debug","msg":"Flags","origincert":"C:\\Windows\\system32\\config\\systemprofile\\.cloudflared\\cert.pem","proxy-dns":true,"proxy-dns-upstream":"https://1.1.1.1/dns-query, https://1.0.0.1/dns-query","time":"2019-09-29T22:44:56+09:00"}
{"addr":{"IP":"127.0.0.1","Port":65453,"Zone":""},"level":"info","msg":"Starting metrics server","time":"2019-09-29T22:44:56+09:00"}
```

**Comment by vidarak** [comment]
Having the same issue on many servers.

**Comment by daltoniam** [comment]
A fix was added in version 2020.8.2 for this bug. Please let us know if upgrading to 2020.8.2 resolves the issue. Thanks!

**Comment by tavrez** [comment]
Fixed in my side, but now there is another weird problem:
If I run cloudflared.exe by double clicking on it, it wipes my config.yml file(with a {} inside it)
but service is ok, no config wipe, and can be stopped correctly
